File: android/src/main/java/org/ligi/passandroid/model/comparator/PassByTypeFirstAndTimeSecondComparator.java
package org.ligi.passandroid.model.comparator;

import java.util.Comparator;
import org.ligi.passandroid.model.pass.Pass;
import static org.ligi.passandroid.model.comparator.DirectionAwarePassByTimeComparator.DIRECTION_ASC;

public class PassByTypeFirstAndTimeSecondComparator implements Comparator<Pass> {

    private final DirectionAwarePassByTimeComparator passByTimeComparator = new DirectionAwarePassByTimeComparator(DIRECTION_ASC);

    @Override
    public int compare(Pass lhs, Pass rhs) {
        final int compareResult = lhs.getType().compareTo(rhs.getType());

        if (compareResult != 0) {
            return compareResult;
        } else {
            return passByTimeComparator.compare(lhs, rhs);
        }
    }
}

File: android/src/main/java/org/ligi/passandroid/model/comparator/PassByTypeFirstAndTimeSecondComparator.kt
package org.ligi.passandroid.model.comparator

import org.ligi.passandroid.model.comparator.DirectionAwarePassByTimeComparator.Companion.DIRECTION_ASC
import org.ligi.passandroid.model.pass.Pass
import java.util.*

class PassByTypeFirstAndTimeSecondComparator : Comparator<Pass> {

    private val passByTimeComparator = DirectionAwarePassByTimeComparator(DIRECTION_ASC)

    override fun compare(lhs: Pass, rhs: Pass): Int {
        val compareResult = lhs.type.compareTo(rhs.type)

        return if (compareResult != 0) {
            compareResult
        } else {
            passByTimeComparator.compare(lhs, rhs)
        }
    }
}
